sourcetracker/_sourcetracker.py

```diff
@@ -9,195 +9,203 @@
 # ----------------------------------------------------------------------------
 from __future__ import division
 
-import os
-from copy import copy
 import numpy as np
 from skbio.stats import subsample_counts
 import pandas as pd
 from functools import partial
-from biom.table import Table
-import glob
-from tempfile import TemporaryDirectory
 
 
-def parse_mapping_file(mf_lines):
-    """Parse lines from a mapping file into a nested dictionary.
+def check_and_correct_data(feature_table, cfc):
+    '''Check and correct problems in the `feature_table`.
 
-    This function copies qiime.parse_mapping_file_to_dict, but does _not_ allow
-    commented lines at the top of a mapping file.
+    Parameters
+    ----------
+    feature_table : pd.DataFrame
+        Contingency table with rows, columns = samples, features.
+    cfc : boolean
+        Fractional counts cause errors in indexing. Uses np.ceil to remove
+        them if True.
+
+    Returns
+    -------
+    feature_table : pd.DataFrame
+        Corrected feature table if changes have been made or original table if
+        no changes occurred. Will automatically be cast as dtype=np.int64
+
+    Raises
+    ------
+    ValueError
+        If data contains values that Gibb's sampling cannot handle.
+    '''
+    _ft = feature_table.copy()
+    if _ft.isnull().values.sum() != 0:
+        raise ValueError('One or more values in the feature table is '
+                         'non-numeric. This will cause propagating error and '
+                         'must be removed.')
+    if not (np.ceil(_ft.values) == _ft.values).all():
+        if cfc is True:
+            _ft.iloc[:, :] = np.ceil(_ft.values)
```

Comment:
I think `np.ceil` is the correct choice because if you have .04 of something that still means you saw it at least once in one of the samples that you used for the averaging. Rounding in that case would say it was never in the source environments. 

I think mean is a better approach than rarifying. The mean is a better estimate of the true proportions (I think, than sum+rarefaction). I can't back that up with statistical reasoning, but it seems like sum+rarefaction tries to accomplish what the mean is doing, it just adds another source of variance (the prng). 

I don't have a strong feeling on rounding vs. ceil, so I'll change it if the consensus is rounding.